File: app/src/main/java/com/example/school/NoticeFragment.java
package com.example.school;

import android.os.AsyncTask;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.content.ContextCompat;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.school.item.NoticeItemModel;

import org.jsoup.Connection;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class NoticeFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_notice, container, false);

        final RecyclerView recyclerView = view.findViewById(R.id.noticeFragment_recyclerview);
        recyclerView.setLayoutManager(new LinearLayoutManager(inflater.getContext()));
        recyclerView.setAdapter(new NoticeFragmentRecyclerView());

        return view;
    }

    class NoticeFragmentRecyclerView extends RecyclerView.Adapter<RecyclerView.ViewHolder>{

        List<NoticeItemModel> noticeModels;
        public NoticeFragmentRecyclerView(){
            try {
                noticeModels = new ArrayList<>();
                noticeModels.clear();
                Connection.Response response = Jsoup.connect("http://school.busanedu.net/daesin-m/na/ntt/selectNttList.do?mi=618566&bbsId=1011229&&listCo=20")
                        .method(Connection.Method.GET)
                        .execute();
                Document document = response.parse();

                Elements noticeItems = document.select("tbody tr");
                for(Element noticeItem: noticeItems) {
                    NoticeItemModel noticeItemModel = new NoticeItemModel();

                    noticeItemModel.title = noticeItem.select("td:nth-child(2)").text();
                    noticeItemModel.writer = noticeItem.select("td:nth-child(3)").text();
                    noticeItemModel.date = noticeItem.select("td:nth-child(4)").text();
                    noticeItemModel.url = noticeItem.select("a").attr("href");
                    noticeItemModel.file_exists = noticeItem.select("img").hasAttr("src");
                    noticeItemModel.isimportant = noticeItem.select("td:nth-child(1)").text() == "공지";
                    noticeModels.add(noticeItemModel);

                    System.out.println(noticeItem);
                }
            } catch (IOException e) {
                e.printStackTrace();
            }
        }

        @NonNull
        @Override
        public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_notice, parent, false);

            return new CustomViewHolder(view);
        }

        @Override
        public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
            ((CustomViewHolder)holder).noticeTitle.setText(noticeModels.get(position).title);
            ((CustomViewHolder)holder).noticeWriter.setText(noticeModels.get(position).writer);
            ((CustomViewHolder)holder).noticeDate.setText(noticeModels.get(position).date);
            if (noticeModels.get(position).file_exists) {
                ((CustomViewHolder)holder).noticeFileExists.setVisibility(View.VISIBLE);
            } else {
                ((CustomViewHolder)holder).noticeFileExists.setVisibility(View.GONE);
            }
            System.out.println("onBindViewHolder");
        }

        @Override
        public int getItemCount() {
            return noticeModels.size();
        }

        private class CustomViewHolder extends RecyclerView.ViewHolder {
            public TextView noticeTitle;
            public TextView noticeWriter;
            public TextView noticeDate;
            public ImageView noticeFileExists;

            public CustomViewHolder(View view) {
                super(view);
                noticeTitle = view.findViewById(R.id.itemNotice_title);
                noticeWriter = view.findViewById(R.id.itemNotice_writer);
                noticeDate = view.findViewById(R.id.itemNotice_date);
                noticeFileExists = view.findViewById(R.id.itemNotice_fileExists);

            }
        }
    }

}

File: app/src/main/java/com/daesin/school/NoticeFragment.kt
package com.daesin.school

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.daesin.school.noticeAdapter.NoticeAdapater
import com.daesin.school.noticeAdapter.NoticeData
import kotlinx.android.synthetic.main.fragment_notice.*
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import org.jsoup.Jsoup

class NoticeFragment : Fragment() {
    //공지사항 객체 리스트
    private val noticeList: ArrayList<NoticeData> = ArrayList()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val holder = inflater.inflate(R.layout.fragment_notice, container, false)
        parse()
        return holder
    }

    private fun parse() {
        //코루틴은 asyncTask랑 비슷한 역활을해줌 비동기 코드 동작
        GlobalScope.launch(Dispatchers.IO) {
            //50개 가져옴
            val jsoup = Jsoup.connect("http://school.busanedu.net/daesin-m/na/ntt/selectNttList.do?mi=618566&bbsId=1011229&&listCo=50").get()
            val document = jsoup.select("tbody tr")
            for (tr in document) {
                //차례대로 제목, 작성자, 날짜, 첨부파일, 링크 확인
                noticeList.add(NoticeData(title = tr.select("td:nth-child(2)").text(),
                        writer = tr.select("td:nth-child(3)").text(),
                        date = tr.select("td:nth-child(4)").text(),
                        file = tr.select("img").hasAttr("src"),
                        link = tr.select("a").attr("href")))
            }

            /**코루틴 상태에서 바로 연결시켜버리면 에러뜸 쓰레드가 필요
             * extensions사용으로 findViewId 해줄필요가 없음
             */
            activity?.runOnUiThread {
                notice_view.layoutManager = LinearLayoutManager(requireContext())
                notice_view.adapter = NoticeAdapater(noticeList)
                noti_loading.visibility = View.GONE
                notice_view.visibility = View.VISIBLE
            }
        }
    }
}